### PR TITLE
Add docs for adothealth monitoring

### PR DESCRIPTION
**Pull request by RAMathews**
### What does this PR do?

This PR is for adding the docs for ADOT collector monitoring.

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted. Consult the CONTRIBUTING guide for submitting pull-requests.

### Motivation

Part of the virtual Hackathon on the [AWS Observability Accelerator](https://github.com/aws-observability/terraform-aws-observability-accelerator).

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [ ] Yes, I ran `pre-commit run -a` with this PR
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-observability/terraform-aws-observability-accelerator/tree/main/examples) to support my PR (when applicable)
- [ ] Yes, I have updated the [Pages](https://github.com/aws-observability/terraform-aws-observability-accelerator/tree/main/docs) for this feature

**Note**: Not all the PRs required examples and docs.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: docs/adothealth/index.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Monitoring ADOT collector health
+
+The OpenTelemetry collector produces metrics to monitor the entire pipeline. In the [EKS monitoring module](https://github.com/aws-observability/terraform-aws-observability-accelerator/tree/main/modules/eks-monitoring), we have enabled those metrics by default with the AWS Distro for OpenTelemetry(ADOT) collector. You get a Grafana dashboard named `OpenTelemetry Health Collector`. This dashboard shows useful telemetry information about the ADOT collector itself which can be helpful when you want to troubleshoot any issues with the collector or understand how much resources the collector is consuming.
```
Comment (lewinkedrs) [inline docs/adothealth/index.md:3]: For the [EKS monitoring] link can we link to the docs page instead of to the github?

Comment (bonclay7) [inline docs/adothealth/index.md:3]: can you add a little space between OpenTelemetry and (ADOT)?

File: docs/adothealth/index.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+!!! note
```
Comment (bonclay7) [inline docs/adothealth/index.md:5]: should be `!!!note`

File: docs/adothealth/index.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+In this dashboard, there are five sections. Each section has [metrics](https://aws-observability.github.io/observability-best-practices/guides/operational/adot-at-scale/operating-adot-collector/#collecting-health-metrics-from-the-collector) relevant to the various [component’s](https://opentelemetry.io/docs/demo/collector-data-flow-dashboard/#data-flow-overview) of the AWS Distro for OpenTelemetry(ADOT)collector:
+
+### Receivers
+Shows the receiver’s accepted and refused rate/count of spans and metric points that are pushed into the telemetry pipeline.
+
+### Processors
+Shows the accepted and refused rate/count of spans and metric points pushed into next component in the pipeline. The batch metrics can help to understand how often metrics are sent to exporter and the batch size.
+
+![recievers_processors](https://github.com/RAMathews/terraform-aws-observability-accelerator/assets/114662591/9a2edc27-9472-4a58-a244-d69f2bc7f41f)
```
Comment (bonclay7) [inline docs/adothealth/index.md:20]: typo on receivers

File: docs/adothealth/index.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+### Exporters
+Shows the exporter’s accepted and refused rate/count of spans and metric points that are pushed to any of the destinations. It also shows the size and capacity of the retry queue. These metrics can be used to understand if the collector is having issues in sending trace or metric data to the destination configured.
+
+![exporters](https://github.com/RAMathews/terraform-aws-observability-accelerator/assets/114662591/77e20ac5-64bb-42ca-9db6-4d13ca7b27de)
+
+### Collectors
+Shows the collector’s operational metrics(Memory, CPU, uptime). This can be used to understand how much resources the collector is consuming.
```
Comment (bonclay7) [inline docs/adothealth/index.md:28]: space after metrics

File: docs/adothealth/index.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+![collectors](https://github.com/RAMathews/terraform-aws-observability-accelerator/assets/114662591/25151edd-6132-479a-9331-71aa69a91d5e)
+
+### Data Flow
+Shows the metrics and spans data flow through the collector’s components.
+
+![dataflow](https://github.com/RAMathews/terraform-aws-observability-accelerator/assets/114662591/61fe684d-8ed3-4645-9210-f16158442b7d)
+
+!!! note
```
Comment (bonclay7) [inline docs/adothealth/index.md:37]: remove space

**Review by lewinkedrs — CHANGES_REQUESTED** [review]
Left one comment on linking to the doc instead of github. Once we do that we can merge. Looks good!

**Review by bonclay7 — COMMENTED** [review]
Few more comments

File: docs/adothealth/index.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+!!!note
+    The dashboard and metrics used are not specific to Amazon EKS, but applicable to any environment running an OpenTelemetry collector.
+
+Below diagram shows an example data flow and the components in an ADOT collector:
+
+![ADOTCollectorComponents](https://github.com/RAMathews/terraform-aws-observability-accelerator/assets/114662591/1db25d84-c1ca-4468-bb0d-42c8bafd1942)
+
+In this dashboard, there are five sections. Each section has [metrics](https://aws-observability.github.io/observability-best-practices/guides/operational/adot-at-scale/operating-adot-collector/#collecting-health-metrics-from-the-collector) relevant to the various [component’s](https://opentelemetry.io/docs/demo/collector-data-flow-dashboard/#data-flow-overview) of the AWS Distro for OpenTelemetry(ADOT)collector:
```
Comment (lewinkedrs) [inline docs/adothealth/index.md:12]: remove the apostrophe it should be "components"

Comment (lewinkedrs) [reply docs/adothealth/index.md:12]: also add a space at the end "AWS Distro for OpenTelemetry(ADOT) collector"

File: docs/adothealth/index.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+![collectors](https://github.com/RAMathews/terraform-aws-observability-accelerator/assets/114662591/25151edd-6132-479a-9331-71aa69a91d5e)
+
+### Data Flow
+Shows the metrics and spans data flow through the collector’s components.
+
+![dataflow](https://github.com/RAMathews/terraform-aws-observability-accelerator/assets/114662591/61fe684d-8ed3-4645-9210-f16158442b7d)
+
+!!!note
+    To read more about the metrics used, and the dashboard use, please visit the upstream documentation [here](https://opentelemetry.io/docs/demo/collector-data-flow-dashboard/).
```
Comment (lewinkedrs) [inline docs/adothealth/index.md:38]: remove the word "please"

**Review by lewinkedrs — CHANGES_REQUESTED** [review]
Spotted a few more typos. Fix these and we should be good to go.

**Review by lewinkedrs — APPROVED** [review]
LGTM!

**Pull request merged**